pkg/fs_algo/fs_algo/fs_perf_viz.py

```diff
@@ -0,0 +1,428 @@
+    dir_std_base = list([x for x in attr_cfg['file_io'] if 'dir_std_base' in x][0].values())[0]
+    dir_std_base = f'{dir_std_base}'.format(dir_base = dir_base)
+    # Options for getting ds_type from a config file:
+    # ds_type = viz_cfg.get('ds_type') # prediction config file IF VISUALIZING PREDICTIONS; attribute config file IF AND ONLY IF VISUALIZING ATTRIBUTES
+    # ds_type = list([x for x in attr_cfg['file_io'] if 'ds_type' in x][0].values())[0]
+    # ...but for plotting purposes, we want to use the prediction ds_type:
+    ds_type = 'prediction'
+    write_type = list([x for x in attr_cfg['file_io'] if 'write_type' in x][0].values())[0]
+
+    # Get features from the main config file --------------------------
+    # NOTE: This assumes that the main config file is just called [same prefix as all other config files]_config.yaml
+    prefix_viz = str(path_viz_config.name).split('_')[0]
+    prefix_attr = str(path_attr_config.name).split('_')[0]
+    if (prefix_viz != prefix_attr):
+        raise ValueError('The base config file (e.g. [dataset]_config.yaml) must be in the same direcotry and identifiable using the same prefix as the other config files (e.g. [dataset]_pred_config.yaml, [dataset]_attr_config.yaml, etc.)')
+    else:
+        prefix = prefix_viz
+
+    path_main_config = fsate.build_cfig_path(path_viz_config,f'{prefix_viz}_config.yaml')
+    with open(path_main_config, 'r') as file:
+        main_cfg = yaml.safe_load(file)
+
+    # NOTE: This is something I'm not totally sure will function properly with multiple datasets
+    formulation_id = list([x for x in main_cfg['formulation_metadata'] if 'formulation_id' in x][0].values())[0]
+    save_type = list([x for x in main_cfg['file_io'] if 'save_type' in x][0].values())[0]
+    if save_type.lower() == 'netcdf':
+        save_type_obs = 'nc'
+        engine = 'netcdf4'
+    else:
+        save_type_obs = 'zarr'
+        engine = 'zarr'
+
+    # Access the location metadata for prediction sites
+    path_meta_pred = pred_cfg.get('path_meta')
+
+    # Location for accessing existing outputs and saving plots
+    dir_out = fsate.fs_save_algo_dir_struct(dir_base).get('dir_out')
+
+    # Loop through all datasets
+    for ds in datasets:
+        path_meta_pred = f'{path_meta_pred}'.format(ds = ds, dir_std_base = dir_std_base, ds_type = ds_type, write_type = write_type)
+        meta_pred = pd.read_parquet(path_meta_pred)
+
+        # Loop through all algorithms
+        for algo in algos:
+            # Loop through all metrics
+            for metric in metrics:
+                # Pull the predictions
+                path_pred = fsate.std_pred_path(dir_out,algo=algo,metric=metric,dataset_id=ds)
+                pred = pd.read_parquet(path_pred)
+                data = pd.merge(meta_pred, pred, how = 'inner', on = 'comid')
+                os.makedirs(f'{dir_out}/data_visualizations', exist_ok= True)
```

Comment:
Making a directory should be a function. For consistency with the rest of RaFTS python, we've been using `pathlib.Path` instead of `os`.  Here's an example of what you're trying to do there from the `std_algo_path`. We could probably use a `std_viz_path()` function.

```
    Path(dir_out_alg_ds).mkdir(exist_ok=True,parents=True)
    basename_alg_ds_metr = f'algo_{algo}_{metric}__{dataset_id}'
    path_algo = Path(dir_out_alg_ds) / Path(basename_alg_ds_metr + '.joblib')
    return path_algo
```